pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/groupby/DictionaryBasedGroupKeyGenerator.java

```diff
@@ -587,6 +617,20 @@ public LongMapBasedHolder(Long2IntOpenHashMap groupIdMap) {
 
     @Override
     public void processSingleValue(int numDocs, int[] outGroupIds) {
+      if (_numGroupByExpressions == 1) {
```

Comment:
We should never get into `LongMapBasedHolder` for single group-by expression

Reply:
Ok I will roll this back, the change was motivated by the int cases I found in profiles and I added this for consistency.